### ISSUE TITLE
Agent and Contact list view not able to shown

### ISSUE THREAD

**Issue by jecinthamary** [frappe/helpdesk#1284]
1. Agent and Contact list not able to view

pymysql.err.OperationalError: (1054, "Unknown column 'user.full_name' in 'field list'")

![image](https://github.com/frappe/helpdesk/assets/126246621/07d4ab21-4061-471d-9f1a-24324caaf999)

**Comment by ssiyad** [comment]
`User` is supposed to have a full name field. What is your Frappe framework version?

**Comment by jecinthamary** [comment]
ERPNext: v14.27.9 (version-14)
Frappe Framework: v14.40.0 (version-14)
Helpdesk: v0.9.1 (main)

![image](https://github.com/frappe/helpdesk/assets/126246621/41e35f2e-8126-4879-9480-80fb1e37d48e)

**Comment by ssiyad** [comment]
I can see `Full Name` field. Can you confirm if the field name is `full_name`? Can you check if such a field exist in DB?

**Comment by jecinthamary** [comment]
User full name is present in DB level, and I installed fresh version 14, but the same error occurred.

**Comment by kurogeek** [comment]
After I did some investigate I found that the errors caused by following SQL Builder produces the wrong output.

Contact: SQL looks like this
```sql
SELECT 
`name`,`email_id`,`image`,`phone`,`tabContact`.`first_name`,`tabContact`.`last_name`,`tabContact Email`.`email_id`,`tabContact Phone`.`phone`,`tabDynamic Link`.`link_name` 
FROM `tabContact` 
LEFT JOIN `tabContact Email` 
    ON `tabContact Email`.`parent`=`tabContact`.`name` 
LEFT JOIN `tabContact Phone` 
    ON `tabContact Phone`.`parent`=`tabContact`.`name` 
LEFT JOIN `tabDynamic Link` 
    ON `tabDynamic Link`.`parent`=`tabContact`.`name` 
ORDER BY `tabContact`.`modified` DESC,`tabContact`.`modified` DESC LIMIT 20
```
As you can see name, email_id, image, and phone fields are ambiguous.

Agent: SQL looks like this.
```sql
SELECT `name`,`is_active`,`user.full_name`,`user.user_image`,`user.email`,`user.username` 
FROM `tabHD Agent` 
ORDER BY `modified` DESC,`modified` DESC LIMIT 20
```
As you can see there are `user.full_name`, `user.user_image`, etc, which `user` has never been defined. And, there are none of those fields inside `tabHD Agent` at all.

**Comment by safvanhuzain** [comment]
Is there any solution for that?

**Comment by kurogeek** [comment]
PR Created on #1308, please check it out.

**Comment by nextchamp-saqib** [comment]
Should be fixed with 24937ad
